packages/plugin-drizzle/src/index.ts

```diff
@@ -1,5 +1,242 @@
-import { DrizzleStorageError } from "./utils";
+import { useIndexerContext } from "@apibara/indexer";
+import { defineIndexerPlugin } from "@apibara/indexer/plugins";
 
-export function drizzleStorage<TFilter, TBlock>() {
-  throw new DrizzleStorageError("Not implemented");
+import type {
+  ExtractTablesWithRelations,
+  TablesRelationalConfig,
+} from "drizzle-orm";
+
+import type { Cursor, DataFinality } from "@apibara/protocol";
+import type {
+  PgDatabase,
+  PgQueryResultHKT,
+  PgTransaction,
+} from "drizzle-orm/pg-core";
+import {
+  finalizeState,
+  getState,
+  initializePersistentState,
+  invalidateState,
+  persistState,
+} from "./persistence";
+import {
+  finalize,
+  initializeReorgRollbackTable,
+  invalidate,
+  registerTriggers,
+  removeTriggers,
+} from "./storage";
+import { DrizzleStorageError, withTransaction } from "./utils";
+
+const DRIZZLE_PROPERTY = "_drizzle";
+
+export type DrizzleStorage<
+  TQueryResult extends PgQueryResultHKT,
+  TFullSchema extends Record<string, unknown> = Record<string, never>,
+  TSchema extends
+    TablesRelationalConfig = ExtractTablesWithRelations<TFullSchema>,
+> = {
+  db: PgTransaction<TQueryResult, TFullSchema, TSchema>;
+};
+
+export function useDrizzleStorage<
+  TQueryResult extends PgQueryResultHKT,
+  TFullSchema extends Record<string, unknown> = Record<string, never>,
+  TSchema extends
+    TablesRelationalConfig = ExtractTablesWithRelations<TFullSchema>,
+>(
+  _db: PgDatabase<TQueryResult, TFullSchema, TSchema>,
```

Comment:
Can we make this parameter optional? The schema is only needed to access the db like the following:

```ts
db.query.users.findMany({ ... });
```

as opposed to 

```ts
db.select().from(users).where(...)
```

Reply:
Alright will make it optional.